src/index.ts

```diff
@@ -147,19 +182,19 @@ function main() {
     if (shouldQuit) { app.quit(); }
 
     // store helps us store data on local disk
-    store = new Store({ name: "explorook" });
-    exceptionManagerEnabled = store.get("sentry-enabled", true);
-    if (exceptionManagerEnabled && !process.env.development) {
-        initExceptionManager("production", app.getVersion());
-    }
+    store = new ExplorookStore();
     // access token used to access this app's GraphQL api
-    token = store.get("token", null);
-    // if first run - there's no token in store - and we create one
-    if (!token) {
+    token = store.getOrCreate("token", uuidv4(), () => {
         firstTimeLaunch = true;
-        token = uuidv4();
-        store.set("token", token);
+    });
+    userId = store.getOrCreate("user-id", uuidv4());
+    exceptionManagerEnabled = store.get("sentry-enabled", true);
+    const signedEula = store.get("has-signed-eula", false);
+    if (signedEula && exceptionManagerEnabled && !process.env.development) {
```

Comment:
we may want to rename `exceptionManagerEnabled` to something like `allowCollectingData`

Reply:
OK